lib/ASTGen/CMakeLists.txt

```diff
@@ -1,20 +1,21 @@
 add_subdirectory(Sources)
 
-# If requested, build the regular expression parser into the compiler itself.
-if(SWIFT_BUILD_REGEX_PARSER_IN_COMPILER)
-  file(GLOB_RECURSE _COMPILER_REGEX_PARSER_SOURCES
-    "${SWIFT_PATH_TO_STRING_PROCESSING_SOURCE}/Sources/_RegexParser/*.swift")
-  set(COMPILER_REGEX_PARSER_SOURCES)
-  foreach(source ${_COMPILER_REGEX_PARSER_SOURCES})
-    file(TO_CMAKE_PATH "${source}" source)
-    list(APPEND COMPILER_REGEX_PARSER_SOURCES ${source})
-  endforeach()
-  message(STATUS "Using Experimental String Processing library for _CompilerRegexParser (${SWIFT_PATH_TO_STRING_PROCESSING_SOURCE}).")
-
-  add_pure_swift_host_library(_CompilerRegexParser STATIC
-    "${COMPILER_REGEX_PARSER_SOURCES}"
-  )
-else()
-  # Dummy target for dependencies
-  add_custom_target(_CompilerRegexParser)
+if(NOT SWIFT_BUILD_REGEX_PARSER_IN_COMPILER)
```

Comment:
I think we may want to allow the case where `SWIFT_BUILD_SWIFT_SYNTAX` is false, as `add_pure_swift_host_library` will bail without creating a target, and it looks like we force `SWIFT_BUILD_REGEX_PARSER_IN_COMPILER` off in that case:

https://github.com/swiftlang/swift/blob/7ea2b3dfecb88659dc7ba4566ec2c26211c11ca5/CMakeLists.txt#L815-L818

cc @rintaro